### PR TITLE
CMake: get rid of Makefile and Ninja

### LINKED ISSUE #153: Pure CMake install not installing most project assets
Following discussion in #151 I've been trying to switch Arch Linux packaging from Meson to CMake. With v1.3.0 sources I'm getting surprising results from `cmake --install`. I'm trying to use the pure CMake backend (not the Ninja backend used in the Makefile.

It's "working" as in running sucessfully, but the only thing it ends up installing is

a) all the sdbuss-c++ stuff (addressed in #151)
b) /usr/bin/hyprland-share-picker

Missing is the systemd service file, the dbus desktop file, the package config file, the config file, and the actual portal. Those things seem to be built with `cmake --build`,, but not installed using `DESTDIR="..." cmake --install <build_dir>`.

### PR DESCRIPTION
**Pull request by fufexan**
Fixes #153.

Addresses all the remaining points in the above issue.
@alerque mind testing this PR?

### CODE REVIEW

**Comment by vaxerski** [comment]
you should also update README for the new instructions. 

I'd generally stick to libexec being in lib/ as from my experience on arch libexec/ isnt picked up.

`-DCMAKE_INSTALL_LIBEXECDIR=${PREFIX}/lib` should do the job, I think

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,7 +93,14 @@ endfunction()
...
+protocol("${HYPRLAND_PROTOCOLS}/protocols/hyprland-toplevel-export-v1.xml" "hyprland-toplevel-export-v1" true)
 protocol("unstable/linux-dmabuf/linux-dmabuf-unstable-v1.xml" "linux-dmabuf-unstable-v1" false)
-##
+
+# Installation
+install(TARGETS hyprland-share-picker)
+install(TARGETS xdg-desktop-portal-hyprland DESTINATION ${CMAKE_INSTALL_LIBEXECDIR})
+
+install(FILES hyprland.portal DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/xdg-desktop-portal/portals")
+install(FILES build/org.freedesktop.impl.portal.desktop.hyprland.service DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/dbus-1/services")
```
Comment (vaxerski) [inline CMakeLists.txt:105]: `build/` -> `${CMAKE_BINARY_DIR}/`

**Comment by alerque** [comment]
It looks like some of the changes here were *intended* to make it possible to build and install hyprland-protocols separately and depend on it from the system (which is a good thing!), but somehow that's managed to break the current submodule arrangement. I haven't tried to build it separately yet (although that will be the preferred way to package this in the long term), but the previous CMake incantation that was finding and building it *stopped* working when run on this PR.

**Comment by fufexan** [comment]
I think it's the `add_subdirectory` call that makes it not work, but could you give me some logs?

> It looks like some of the changes here were _intended_ to make it possible to build and install hyprland-protocols separately and depend on it from the system

Yes. Maybe I've said it before, but I hate git submodules :smile:

**Comment by alerque** [comment]
That latest force-push is a *lot* better.

Now the only differences between the previous release packaged for Arch (using Meson) and my current attempts with 1.3.0 plus this PR using generic CMake are that some of the hyprland-protocols things are not being installed. The subdirectory build does work to satisfy the dependency but it does not end up installing the pc file or protocol files for it. I haven't tested yet whether those are needed at run time. This result might actually be expected & correct in that those should only be vendored if using the subproject and only installed to the system if built and installed in their own right.

If the latter is the expected behavior then this PR might be :+1: go for launch.

**Comment by fufexan** [comment]
> This result might actually be expected & correct in that those should only be vendored if using the subproject and only installed to the system if built and installed in their own right.

Yes, the .pc file and the XML protocols are only needed when `hyprland-protocols` is a standalone package.

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**